src/Result.ts

```diff
@@ -3,104 +3,169 @@ egjs-list-differ
 Copyright (c) 2019-present NAVER Corp.
 MIT license
 */
-import Link from "./Link";
+import {
+  AfterOrderIndex,
+  Change,
+  CurrentIndex,
+  Order,
+  PrevIndex,
+} from "./types";
 
-function orderChanged(changed: number[][], fixed: boolean[]) {
-  // It is roughly in the order of these examples.
-  // 4, 6, 0, 2, 1, 3, 5, 7
-  const fromLinks: Link[] = [];
-  // 0, 1, 2, 3, 4, 5, 6, 7
-  const toLinks: Link[] = [];
+function LIS(arr: number[]) {
+  if (arr.length === 0) {
+    return [];
+  }
 
-  changed.forEach(([from, to]) => {
-    const link = new Link();
+  const table: number[] = [];
+  const path: number[] = [];
+  arr.forEach((_, i) => {
+    table[i] = 1;
+    path[i] = -1;
+  });
 
-    fromLinks[from] = link;
-    toLinks[to] = link;
+  arr.forEach((_, i) => {
+    for (let j = i - 1; j > -1; j -= 1) {
+      if (table[j] + 1 > table[i] && arr[j] <= arr[i]) {
+        table[i] = table[j] + 1;
+        path[i] = j;
+      }
+    }
   });
-  // `fromLinks` are connected to each other by double linked list.
-  fromLinks.forEach((link, i) => {
-    link.connect(fromLinks[i - 1]);
+
+  let maxIndex = 0;
+  table.forEach((x, i) => {
+    if (x > table[maxIndex]) {
+      maxIndex = i;
+    }
   });
 
-  return changed.filter((_, i) => !fixed[i]).map(([from, to], i) => {
-    if (from === to) {
-      return [0, 0];
+  const res: number[] = [];
+  while (maxIndex !== -1) {
+    res.push(arr[maxIndex]);
+    maxIndex = path[maxIndex];
+  }
+
+  return res.reverse();
+}
+
+function orderChanged(priorityList: AfterOrderIndex[], maintained: Change[]) {
+  const confirmed: Record<number, boolean> = {};
+  const ordered: Order[] = [];
+  const length = priorityList.length;
+
+  LIS(priorityList).forEach((x) => {
+    confirmed[x] = true;
+  });
+
+  for (let from = 0; from < length; from += 1) {
+    const priority = priorityList[from];
+    if (confirmed[priority]) {
+      continue;
     }
-    const fromLink = fromLinks[from];
-    const toLink = toLinks[to - 1];
-    const fromIndex = fromLink.getIndex();
-
-    // Disconnect the link connected to `fromLink`.
-    fromLink.disconnect();
-
-    // Connect `fromLink` to the right of `toLink`.
-    if (!toLink) {
-      fromLink.connect(undefined, fromLinks[0]);
-    } else {
-      fromLink.connect(toLink, toLink.next);
+
+    let to = 0;
+    while (to < length) {
+      if (confirmed[priorityList[to]] && priorityList[to] > priority) {
+        break;
+      }
+      to += 1;
     }
-    const toIndex = fromLink.getIndex();
-    return [fromIndex, toIndex];
-  });
+    to < from && (to += 1);
+
+    confirmed[priority] = true;
+    ordered.push([
+      from,
+      to - 1,
+      maintained[priority][0],
+      to < length ? maintained[priorityList[to]][1] : -1,
```

Comment:
[from, to], [prev, current?], anchor mismatched. The current index does not exist.

Lots of tuple information. It is better to express it as an object.